### ISSUE TITLE
Need help with debugging

### ISSUE THREAD

**Issue by mwhiter** [LoneGazebo/Community-Patch-DLL#1275]
Sorry for such a shitty title. I wasn't sure how to label this. :( 

Hey, so I'm experiencing a division by zero bug on the Happiness Tooltip (which is instant CTD). I am currently using a CP-only configuration for developing C4DF's update.

The problem function is CvLuaPlayer::GetLuxuryBonusPlusOne(). The value of getBALANCE_HAPPINESS_LUXURY_BASE() is 0 instead of 8, which causes immediate CTD because of division by zero. It's called in the Happiness tooltip which is what's causing the issue, so to fix I'm going to need heavy modifications of those LUA files.

How can I disable MOD_BALANCE_CORE and associated C++ defines in the DLL? (I think this should be done from the XML but I'm having trouble disabling them...)

**Comment by LoneGazebo** [comment]
Don't disable MOD_BALANCE_CORE! That's absolutely essential for everything. Easiest solution would be to have it default to 1 in the DLL.

I can also set that function (GetLuxuryBonusPlusOne) to have a min of 1.

G

**Comment by mwhiter** [comment]
Thanks. I won't. However, I believe  that function's global is defined in Civ Balance Patch, so something is wrong there if anything from Community Patch is calling it. I commented out the problem LUA call but the function itself is the issue. 

Or perhaps I'm trying to do something stupid? Maybe I should heavily edit the LUA files I imported from C4DF CBP to be compatible with CP only -   but I really don't know what the CBP functionality  and the CP functionality is.

Nevertheless, I got my build stable again through the commenting. I just figured I'd give a heads up on a potential issue.
